### PR TITLE
topology2: add sof-mtl-rt713-l0-rt1316-l12-rt1713-l3 support

### PR DESCRIPTION
**Pull request by bardliao**
Add sof-mtl-rt713-l0-rt1316-l12-rt1713-l3 support.
This is the topology for https://github.com/thesofproject/linux/pull/4538

This PR has not been tested on a device, yet. Please wait for the test before merging.

### CODE REVIEW

File: tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake [hunk 1]
```diff
@@ -35,6 +35,10 @@ SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack"
 USE_CHAIN_DMA=true,NUM_SDW_AMP_LINKS=2,SDW_SPK_STREAM=SDW1-Playback,SDW_SPK_IN_STREAM=SDW1-Capture,\
 SDW_DMIC_STREAM=SDW0-Capture"
 
+"cavs-sdw\;sof-mtl-rt713-l0-rt1316-l12-rt1713-l3\;PLATFORM=mtl,NUM_SDW_AMP_LINKS=2,SDW_DMIC=1,\
+NUM_HDMIS=3,SDW_SPK_STREAM=Playback-SmartAmp,SDW_SPK_IN_STREAM=Capture-SmartAmp,\
+SDW_DMIC_STREAM=Capture-SmartMic,SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack,\
+PREPROCESS_PLUGINS=nhlt,NHLT_BIN=sof-mtl-rt713-l0-rt1316-l12-rt1713-l3.bin"
```
Comment (plbossart) [inline tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake:41]: do we need any NHLT stuff?

Comment (bardliao) [reply tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake]: > do we need any NHLT stuff?

No, we don't. Thanks for pointing it out.

**Comment by bardliao** [comment]
ODM confirmed that the patch has been verified on a real device. So, it is ready for review now.

File: tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake [hunk 2]
```diff
@@ -35,6 +35,9 @@ SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack"
 USE_CHAIN_DMA=true,NUM_SDW_AMP_LINKS=2,SDW_SPK_STREAM=SDW1-Playback,SDW_SPK_IN_STREAM=SDW1-Capture,\
 SDW_DMIC_STREAM=SDW0-Capture"
 
+"cavs-sdw\;sof-mtl-rt713-l0-rt1316-l12-rt1713-l3\;PLATFORM=mtl,NUM_SDW_AMP_LINKS=2,SDW_DMIC=1,\
+NUM_HDMIS=3,SDW_SPK_STREAM=Playback-SmartAmp,SDW_SPK_IN_STREAM=Capture-SmartAmp,\
```
Comment (plbossart) [inline tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake:39]: I though NUM_HDMIS was set to 3 as a platform default?

I don't know why some platforms has NUM_HDMIS=0 btw, this looks quite wrong now.

Comment (bardliao) [reply tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake]: > I though NUM_HDMIS was set to 3 as a platform default?
> 

Yes, and updated.

> I don't know why some platforms has NUM_HDMIS=0 btw, this looks quite wrong now.

Those are for MTL RVP, and i915 driver is not ready on the sof-dev branch. We will remove those NUM_HDMI=0 once i915 driver is ready.

**Review by plbossart — APPROVED** [review]

File: tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake (hunk 2 above)
Comment (kv2019i) [reply tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake]: @bardliao @plbossart we do need to address this better. The HDA generic driver can handle missing HDMI and the SDW mach drivers should do the same. We can have variants with internal graphics disabled and this should not prevent audio from working on these devices (and we don't want to ship separate tplg variants for such cases).

**Comment by mengdonglin** [comment]
@kv2019i Good to merge? And we can cherry pick to v2.7

**Comment by mengdonglin** [comment]
@kv2019i Do you think it's good to merge now?  And we can cherry pick to v2.7 :)

**Pull request merged**